controller/reserves.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"go-project/model"
+	"io/ioutil"
+	"net/http"
+
+	"go-project/service"
+
+	"github.com/gorilla/mux"
+)
+
+type (
+	ReserveController struct {
+		reserveService *service.ReserveService
+	}
+
+	ReserveControllerImpl interface {
+		//CreateNewBooking(w http.ResponseWriter, r *http.Request)
+	}
+)
+
+func NewController(reserveService *service.ReserveService) *ReserveController {
+	return &ReserveController{
+		reserveService: reserveService,
+	}
+}
```

Comment:
Actually, we don't want to expose the struct outside of package

I mean to protect other dev to create new struct like this:
```golang
 reserveController := ReserveController{}
 
 // This won't work because this struct doesn't contain any fields.
 reserveConrtoller.GetAllTable()
```
It will make it hard for other developers to add new features or fix things.

We use "New" function to validate fields inside structs.
And input of New for this package we want only interface of reserve service.

like this:
```golang
  func New(service service.ReserveServiceImpl) reserveContoller {
       // so, if doesn't have any validate just return it anywhere
       return reserveController{
        service: service,
       }
   }
```